File: Target LeetdCode 200/runner.cpp
#include <cstdlib>
#include <string>
#include <fstream>
#include <iostream>
#include <vector>

using namespace std;

class Node
{
public:
    int data;
    Node *next;
    Node(int data)
    {
        this->data = data;
        this->next = NULL;
    }
    ~Node()
    {
        if (next)
        {
            delete next;
        }
    }
};

#include "solution.h"

class Runner
{
    int t;
    vector<Node*> arr1;
    vector<Node*> arr2;

public:
    void takeInput()
    {
        cin >> t;
        for (int c = 0; c < t; c++)
        {
            int data;
            cin >> data;
            Node *head = NULL, *tail = NULL;
            while (data != -1)
            {
                Node *newNode = new Node(data);
                if (head == NULL)
                {
                    head = newNode;
                    tail = newNode;
                }
                else
                {
                    tail->next = newNode;
                    tail = newNode;
                }
                cin >> data;
            }

            arr1.push_back(head);

            cin >> data;
            head = NULL;
            tail = NULL;
            while (data != -1)
            {
                Node *newNode = new Node(data);
                if (head == NULL)
                {
                    head = newNode;
                    tail = newNode;
                }
                else
                {
                    tail->next = newNode;
                    tail = newNode;
                }
                cin >> data;
            }

            arr2.push_back(head);

        }
    }

    void printList(Node *head)
    {
        Node *tail = head;

        while (tail != NULL && tail->next != head)
        {
            cout << tail->data << ' ';
            tail = tail->next;
        }

        if (tail != NULL)
            cout << tail->data << ' ';
        cout << "-1" << endl;
    }

    Node *copyLinkedList(Node *head)
    {
        if (head == NULL)
        {
            return NULL;
        }
        Node *newLinkedList = new Node(head->data);
        Node *cur = newLinkedList;
        head = head->next;
        while (head != NULL)
        {
            cur->next = new Node(head->data);
            cur = cur->next;
            head = head->next;
        }
        return newLinkedList;
    }

    void execute()
    {
        for (int i = 0; i < t; i++)
        {
            Node *head1 = copyLinkedList(arr1[i]);
            Node *head2 = copyLinkedList(arr2[i]);
            Node *head = addLL(head1, head2);

            Node *prev1 = head1;
            Node *prev2 = head2;

            while (prev1 != NULL)
            {
                Node*temp = prev1->next;
                prev1->next = NULL;
                delete prev1;
                prev1 = temp;

            }

            while (prev2 != NULL)
            {
                Node*temp = prev2->next;
                prev2->next = NULL;
                delete prev2;
                prev2 = temp;

            }

        }
    }

    void executeAndPrintOutput()
    {
        for (int i = 0; i < t; i++)
        {
            Node *head1 = copyLinkedList(arr1[i]);
            Node *head2 = copyLinkedList(arr2[i]);
            Node *head = addLL(head1, head2);
            printList(head);
            Node *prev1 = head;
            while (prev1 != NULL)
            {
                Node*temp = prev1->next;
                prev1->next = NULL;
                delete prev1;
                prev1 = temp;

            }
        }
    }
};

int main()
{
    Runner runner;
    runner.takeInput();
    runner.executeAndPrintOutput();
    return 0;
}

File: Target LeetdCode 200/Runner.java
import java.util.ArrayList;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.io.FileWriter;
import java.io.BufferedReader;
import java.io.InputStreamReader;

class Node {
    public int data;
    public Node next, prev;

    Node(int data) {
        this.data = data;
        this.next = null;
        this.prev = null;
    }
}

public class Runner {

    private static int t;
    private static ArrayList<Node > allLists;
    private static ArrayList<Integer > allPos;
    private static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

    private static Node linkedListInput() throws IOException {
        String [] data;
        data = br.readLine().trim().split(" ");

        Node head = null;
        Node tail = null;
        int index = 0;

        while (Integer.parseInt(data[index]) != -1) {
            Node newNode = new Node(Integer.parseInt(data[index++]));
            if (head == null) {
                head = newNode;
                tail = newNode;
            } else {
                tail.next = newNode;
                newNode.prev = tail;
                tail = tail.next;
            }
        }

        return head;
    }

    private static void takeInput() throws IOException {

        t = Integer.parseInt(br.readLine().trim().split(" ")[0]);

        allLists = new ArrayList<Node>(t);
        allPos = new ArrayList<Integer>(t);

        for (int i = 0; i < t; i++) {
            allLists.add(linkedListInput());
            allPos.add(Integer.parseInt(br.readLine().trim().split(" ")[0]));

        }
    }

    private static Node copyLinkedList(Node head) {
        if (head == null) {
            return null;
        }
        Node newLinkedList = new Node(head.data);
        Node cur = newLinkedList;
        head = head.next;
        while (head != null) {
            cur.next = new Node(head.data);
            cur = cur.next;
            head = head.next;
        }
        return newLinkedList;
    }

    private static void execute() {
        for (int i = 0; i < t; i++) {
            Node head = copyLinkedList(allLists.get(i));

            Node result = Solution.deleteNode(head, allPos.get(i));

        }
    }

    private static void executeAndPrintOutput() {
        for (int i = 0; i < t; i++) {
            Node head = copyLinkedList(allLists.get(i));
            Node ans = Solution.deleteNode(head, allPos.get(i));

            while (ans != null) {
                System.out.print(ans.data + " ");
                ans = ans.next;
            }
            System.out.println(-1);


        }
    }

    public static void main(String[] args) throws IOException {
        takeInput();
        // executeAndPrintOutput();

        int execCount = Integer.parseInt(System.getenv("EXECUTION_COUNT"));

        if (execCount == 1) {
            executeAndPrintOutput();
        } else {

            FileWriter writer = new FileWriter(System.getenv("EXEC_COUNTER_FILE"));

            for (int i = 0; i < execCount; i++) {
                execute();
                writer.write(String.valueOf(i + 1) + "\n");
                writer.flush();
            }

            writer.close();
        }
    }
}